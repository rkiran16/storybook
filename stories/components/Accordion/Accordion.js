import { html } from "lit-html";

export const Accordion = ({isOpen = false, label, collapseContainerId, accordionId, disabled = false, accordionSmall = false, accordionNoBorder = false}) => {
  const show = isOpen ? 'show': '';
  const btnCollapseClass = isOpen ? '': 'collapsed';
  const ariaExpanded = isOpen ? true: false;
  const smallAccordionClass = accordionSmall ? 'accordion-small': '';
  const accordionNoBorderClass = accordionNoBorder ? 'accordion-no-border': ''; 

  return html `
    <div class=${["accordion", smallAccordionClass, accordionNoBorderClass].join(" ")} id=${accordionId}>
      <div class="accordion-item">
        <h5 class="accordion-header" id="headingOne">
          <button class=${["accordion-button", btnCollapseClass].join(" ")} ?disabled=${disabled} type="button" data-bs-toggle="collapse" 
                  data-bs-target=${['#', collapseContainerId].join("")} aria-expanded=${ariaExpanded} aria-controls=${collapseContainerId} >
            ${label}
          </button>
        </h5>
        <div id=${collapseContainerId} class=${["accordion-collapse collapse", show ].join(' ')} 
             aria-labelledby="headingOne" data-bs-parent=${['#', accordionId].join("")}>
          <div class="accordion-body">
             <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce leo mauris, scelerisque quis condimentum sed, blandit convallis lacus. 
                Phasellus consequat, nibh sed vestibulum posuere, ipsum nunc consequat tortor, 
                nec scelerisque ligula dolor nec neque. Integer faucibus nunc non consequat venenatis.
             </p>
          </div>
        </div>
      </div>
    </div>
  `;
}

export const MultipleAccordion = ({disabled = false, accordionSmall = false}) => {
  const smallAccordionClass = accordionSmall ? 'accordion-small': '';

   return html`
   <div class=${["accordion", smallAccordionClass].join(" ")} id="accordionPanelsStayOpenExample">
   <div class="accordion-item">
     <h5 class="accordion-header" id="panelsStayOpen-headingOne">
       <button class="accordion-button" ?disabled=${disabled} type="button" data-bs-toggle="collapse" data-bs-target="#panelsStayOpen-collapseOne" aria-expanded="true" aria-controls="panelsStayOpen-collapseOne">
         Accordion Item #1
       </button>
     </h5>
     <div id="panelsStayOpen-collapseOne" class="accordion-collapse collapse show" aria-labelledby="panelsStayOpen-headingOne">
       <div class="accordion-body">
          Accordion Body Goes Here
       </div>
     </div>
   </div>
   <div class="accordion-item">
     <h5 class="accordion-header" id="panelsStayOpen-headingTwo">
       <button class="accordion-button collapsed" ?disabled=${disabled} type="button" data-bs-toggle="collapse" data-bs-target="#panelsStayOpen-collapseTwo" aria-expanded="false" aria-controls="panelsStayOpen-collapseTwo">
         Accordion Item #2
       </button>
     </h5>
     <div id="panelsStayOpen-collapseTwo" class="accordion-collapse collapse" aria-labelledby="panelsStayOpen-headingTwo">
       <div class="accordion-body">
         Accordion Body Goes Here
       </div>
     </div>
   </div>
   <div class="accordion-item">
     <h5 class="accordion-header" id="panelsStayOpen-headingThree">
       <button class="accordion-button collapsed" ?disabled=${disabled} type="button" data-bs-toggle="collapse" data-bs-target="#panelsStayOpen-collapseThree" aria-expanded="false" aria-controls="panelsStayOpen-collapseThree">
         Accordion Item #3
       </button>
     </h5>
     <div id="panelsStayOpen-collapseThree" class="accordion-collapse collapse" aria-labelledby="panelsStayOpen-headingThree">
       <div class="accordion-body">
          Accordion Body Goes Here
       </div>
     </div>
   </div>
 </div>
   `
}