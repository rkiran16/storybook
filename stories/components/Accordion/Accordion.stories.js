import { Accordion, MultipleAccordion } from "./Accordion";

export default {
   title: "Components/Accordion",
   parameters: {
      layout: 'centered',
      zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62902abd06999bae8aaf71a0",
   },
   argTypes: {
      isOpen: {
         type: { name: 'boolean', required: false},
         description: "Whether or not the accordion is open",
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: {
            type: 'boolean'
         }
      },
      label: {
         name: 'label',
         type: { name: 'string', required: false },
         defaultValue: 'Accordion label',
         description: 'The label of the collapse',
         control: {
           type: 'text'
         }
      },
      disabled: {
        type: { name: 'boolean', required: false},
        description: "Use when the user cannot proceed or until an input is collected",
        defaultValue: false,
        table: {
          type: { summary: 'boolean' },
          defaultValue: { summary: false },
        },
        control: {
          type: 'boolean'
        }
      }
   }
}



const Template = (args) => Accordion(args);
const MultipleAccordionTemplate = (args) => MultipleAccordion(args);

export const Standard = Template.bind({});
Standard.args = {
   accordionId: "AccordionOne",
   collapseContainerId: "AccordionExample"
}

export const AlwaysOpen = Template.bind({});
AlwaysOpen.args = {
   isOpen: true,
   accordionId: "accordionOpen",
   collapseContainerId: "AccordionTwo"
}

export const Small = Template.bind({});
Small.args = {
   accordionId: "accordionSmall",
   collapseContainerId: "AccordionThree",
   accordionSmall: true
}

export const No_Border = Template.bind({});
No_Border.args = {
   accordionId: "accordionNoBorder",
   collapseContainerId: "AccordionNoBorderContainer",
   accordionSmall: true,
   accordionNoBorder: true
}

export const MultipleAccordionItems = MultipleAccordionTemplate.bind({});
MultipleAccordionItems.args = {
   accordionSmall: false
}
