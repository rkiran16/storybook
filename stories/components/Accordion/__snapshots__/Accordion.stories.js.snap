// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Accordion Always Open 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " id=",
    ">
      <div class=\\"accordion-item\\">
        <h5 class=\\"accordion-header\\" id=\\"headingOne\\">
          <button class=",
    " ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" 
                  data-bs-target=",
    " aria-expanded=",
    " aria-controls=",
    " >
            ",
    "
          </button>
        </h5>
        <div id=",
    " class=",
    " 
             aria-labelledby=\\"headingOne\\" data-bs-parent=",
    ">
          <div class=\\"accordion-body\\">
             <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce leo mauris, scelerisque quis condimentum sed, blandit convallis lacus. 
                Phasellus consequat, nibh sed vestibulum posuere, ipsum nunc consequat tortor, 
                nec scelerisque ligula dolor nec neque. Integer faucibus nunc non consequat venenatis.
             </p>
          </div>
        </div>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "accordion  ",
    "accordionOpen",
    "accordion-button ",
    false,
    "#AccordionTwo",
    true,
    "AccordionTwo",
    "Accordion label",
    "AccordionTwo",
    "accordion-collapse collapse show",
    "#accordionOpen",
  ],
}
`;

exports[`Storyshots Components/Accordion Multiple Accordion Items 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
   <div class=",
    " id=\\"accordionPanelsStayOpenExample\\">
   <div class=\\"accordion-item\\">
     <h5 class=\\"accordion-header\\" id=\\"panelsStayOpen-headingOne\\">
       <button class=\\"accordion-button\\" ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#panelsStayOpen-collapseOne\\" aria-expanded=\\"true\\" aria-controls=\\"panelsStayOpen-collapseOne\\">
         Accordion Item #1
       </button>
     </h5>
     <div id=\\"panelsStayOpen-collapseOne\\" class=\\"accordion-collapse collapse show\\" aria-labelledby=\\"panelsStayOpen-headingOne\\">
       <div class=\\"accordion-body\\">
          Accordion Body Goes Here
       </div>
     </div>
   </div>
   <div class=\\"accordion-item\\">
     <h5 class=\\"accordion-header\\" id=\\"panelsStayOpen-headingTwo\\">
       <button class=\\"accordion-button collapsed\\" ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#panelsStayOpen-collapseTwo\\" aria-expanded=\\"false\\" aria-controls=\\"panelsStayOpen-collapseTwo\\">
         Accordion Item #2
       </button>
     </h5>
     <div id=\\"panelsStayOpen-collapseTwo\\" class=\\"accordion-collapse collapse\\" aria-labelledby=\\"panelsStayOpen-headingTwo\\">
       <div class=\\"accordion-body\\">
         Accordion Body Goes Here
       </div>
     </div>
   </div>
   <div class=\\"accordion-item\\">
     <h5 class=\\"accordion-header\\" id=\\"panelsStayOpen-headingThree\\">
       <button class=\\"accordion-button collapsed\\" ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#panelsStayOpen-collapseThree\\" aria-expanded=\\"false\\" aria-controls=\\"panelsStayOpen-collapseThree\\">
         Accordion Item #3
       </button>
     </h5>
     <div id=\\"panelsStayOpen-collapseThree\\" class=\\"accordion-collapse collapse\\" aria-labelledby=\\"panelsStayOpen-headingThree\\">
       <div class=\\"accordion-body\\">
          Accordion Body Goes Here
       </div>
     </div>
   </div>
 </div>
   ",
  ],
  "values": Array [
    "accordion ",
    false,
    false,
    false,
  ],
}
`;

exports[`Storyshots Components/Accordion No Border 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " id=",
    ">
      <div class=\\"accordion-item\\">
        <h5 class=\\"accordion-header\\" id=\\"headingOne\\">
          <button class=",
    " ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" 
                  data-bs-target=",
    " aria-expanded=",
    " aria-controls=",
    " >
            ",
    "
          </button>
        </h5>
        <div id=",
    " class=",
    " 
             aria-labelledby=\\"headingOne\\" data-bs-parent=",
    ">
          <div class=\\"accordion-body\\">
             <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce leo mauris, scelerisque quis condimentum sed, blandit convallis lacus. 
                Phasellus consequat, nibh sed vestibulum posuere, ipsum nunc consequat tortor, 
                nec scelerisque ligula dolor nec neque. Integer faucibus nunc non consequat venenatis.
             </p>
          </div>
        </div>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "accordion accordion-small accordion-no-border",
    "accordionNoBorder",
    "accordion-button collapsed",
    false,
    "#AccordionNoBorderContainer",
    false,
    "AccordionNoBorderContainer",
    "Accordion label",
    "AccordionNoBorderContainer",
    "accordion-collapse collapse ",
    "#accordionNoBorder",
  ],
}
`;

exports[`Storyshots Components/Accordion Small 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " id=",
    ">
      <div class=\\"accordion-item\\">
        <h5 class=\\"accordion-header\\" id=\\"headingOne\\">
          <button class=",
    " ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" 
                  data-bs-target=",
    " aria-expanded=",
    " aria-controls=",
    " >
            ",
    "
          </button>
        </h5>
        <div id=",
    " class=",
    " 
             aria-labelledby=\\"headingOne\\" data-bs-parent=",
    ">
          <div class=\\"accordion-body\\">
             <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce leo mauris, scelerisque quis condimentum sed, blandit convallis lacus. 
                Phasellus consequat, nibh sed vestibulum posuere, ipsum nunc consequat tortor, 
                nec scelerisque ligula dolor nec neque. Integer faucibus nunc non consequat venenatis.
             </p>
          </div>
        </div>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "accordion accordion-small ",
    "accordionSmall",
    "accordion-button collapsed",
    false,
    "#AccordionThree",
    false,
    "AccordionThree",
    "Accordion label",
    "AccordionThree",
    "accordion-collapse collapse ",
    "#accordionSmall",
  ],
}
`;

exports[`Storyshots Components/Accordion Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " id=",
    ">
      <div class=\\"accordion-item\\">
        <h5 class=\\"accordion-header\\" id=\\"headingOne\\">
          <button class=",
    " ?disabled=",
    " type=\\"button\\" data-bs-toggle=\\"collapse\\" 
                  data-bs-target=",
    " aria-expanded=",
    " aria-controls=",
    " >
            ",
    "
          </button>
        </h5>
        <div id=",
    " class=",
    " 
             aria-labelledby=\\"headingOne\\" data-bs-parent=",
    ">
          <div class=\\"accordion-body\\">
             <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce leo mauris, scelerisque quis condimentum sed, blandit convallis lacus. 
                Phasellus consequat, nibh sed vestibulum posuere, ipsum nunc consequat tortor, 
                nec scelerisque ligula dolor nec neque. Integer faucibus nunc non consequat venenatis.
             </p>
          </div>
        </div>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "accordion  ",
    "AccordionOne",
    "accordion-button collapsed",
    false,
    "#AccordionExample",
    false,
    "AccordionExample",
    "Accordion label",
    "AccordionExample",
    "accordion-collapse collapse ",
    "#AccordionOne",
  ],
}
`;
